File: Task3/C/Subtask3/Task3.c
#include <stdio.h>

int main()
{
	int num;
	printf("Enter diamond width: \n");
	scanf("%d", &num);
	
	for(int i=0;i<2*num-1,i++){
		int row;
		if (i<num){
			row=2*i+1;
		}
		else{
			row=2*(2*num-i-1)-1;
		}
		for(int x=0; k<2*n-comp; x++) {
			printf(" ");
		}
		for (j=0;j<2*n-row;j++) {
			print("* ");
		}
	print("\n");
	}
	return 0;
}

File: Task3/rust/Subtask3/Task3.rs
use std::io;

fn main() -> io::Result<()> {
    let mut input = String::new();
    println!("Enter width of diamond: ");
    io::stdin().read_line(&mut input)?;
    let n: usize = input.trim().parse().unwrap();

    for i in 0..n {
        for _ in 0..(n - i - 1) {
            print!(" ");
        }
        for _ in 0..(2 * i + 1) {
            print!("*");
        }
        println!();
    }
    
    for i in (0..n - 1).rev() {
        for _ in 0..(n - i - 1) {
            print!(" ");
        }
        for _ in 0..(2 * i + 1) {
            print!("*");
        }
   

